tests/pipeline_test.go

```diff
@@ -49,7 +49,23 @@ func (suite *IntegrationTestSuite) TestAPIServerDeployment() {
 
 		response, err := http.Post(postUrl, contentType, body)
 		require.NoError(t, err)
-		responseData, err := ioutil.ReadAll(response.Body)
+		responseData, err := io.ReadAll(response.Body)
+		responseString := string(responseData)
+		loggr.Info(responseString)
+		require.NoError(t, err)
+		assert.Equal(t, 200, response.StatusCode)
+	})
+
+	suite.T().Run("Should successfully upload a pipeline with custom pip server", func(t *testing.T) {
+		postUrl := fmt.Sprintf("%s/apis/v2beta1/pipelines/upload", APIServerURL)
+		vals := map[string]string{
+			"uploadfile": "@resources/test-pipeline-with-custom-pip-server-run.yaml",
```

Comment:
Let's set the name explicitly using the `name` parameter so it doesn't use the filename [here](https://github.com/opendatahub-io/data-science-pipelines-operator/pull/634/files#)

You should apply the same change [here](https://github.com/opendatahub-io/data-science-pipelines-operator/pull/634/files#diff-c4ab1c1240f7555966181c6d8a019be37701a2eb479f28c32d962da04ec2e2b6R46) because currently this test does not use this pipeline that is uploaded, but instead uses the [default iris pipeline](https://github.com/opendatahub-io/data-science-pipelines-operator/pull/634/files#diff-fd78d79230e8185a4a9a085795fed899c0fa49fe5e0168e910013a709fc55bf1R37) that deploys with dsp

Reply:
not yet resolved, set to unresolved